staging-service-shared/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/BatchPathProvider.java

```diff
@@ -138,4 +217,13 @@ public UnexpectedInvalidBatchIdException(final BatchId batchId)
             super("Invalid batch id: " + batchId);
         }
     }
+
+    private static void threadSleep()
+    {
+        try {
+            Thread.sleep(1000);
+        } catch (InterruptedException e) {
```

Comment:
We shouldn't be catching this.  It should be allowed to propagate all the way up to `StagingController`.  At that point it should probably be mapped to a `503 Service Unavailable` response.

Reply:
Done.

Reply:
You've caught it early and re-thrown it as a `ServiceUnavailableException`, which you've then re-caught in `StagingController`.  Just propagate the `InterruptedException` please.

Reply:
Propagated to the controller.